src/Microsoft.AspNetCore.Mvc.ViewFeatures/Resources.resx

```diff
@@ -286,4 +286,10 @@
   <data name="ViewEnginesAreRequired" xml:space="preserve">
     <value>'{0}.{1}' must not be empty. At least one '{2}' is required to locate a view for rendering.</value>
   </data>
+  <data name="TempDataProperties_PrimitiveTypeOrString" xml:space="preserve">
+    <value>TempData property {0} is not declared as a primitive type or string.</value>
+  </data>
+  <data name="TempDataProperties_PublicGetterSetter" xml:space="preserve">
+    <value>TempData property {0} does not have a public getter or setter.</value>
```

Comment:
These should be more friendly to read, like:

> The {0} property with {1} is invalid. A property using {1} must have a public getter and setter.

Never hardcode the names of 'code things' like `TempData` in a resource. It's better to make it a parameter and pass in the name.

Make sure that the property name that's reported here includes the full type name as well. We want to make sure that you can find the issue 👍